app/routes.py

```diff
@@ -2,25 +2,75 @@
 from flask import jsonify, request
 from app.models import User, Post
 
-# Create, Update, Get, Delete one post from db
 
-@app.route('/post/<post_id>', methods =['GET'])
+# Create, Update, Get, Delete one post from db
+@app.route('/post/<post_id>', methods =['GET', 'PUT', 'DELETE'])
 def manage_post(post_id):
-    post = Post.query.get(post_id)
-    return jsonify({ 
-     'body':post.body , 
-     'timestamp':post.timestamp,
-     'user_id':post.user_id
-    }), 200
+    if request.method == 'GET':
+        post = Post.query.get(post_id)
+        if post == None :
+            return jsonify({"msg": "there is no post"}), 400
+    
+        else :
+            return jsonify({ 
+            'body':post.body , 
+            'timestamp':post.timestamp,
+            'user_id':post.user_id
+        }), 200
+    
+    elif request.method == 'PUT':
+        data = request.get_json(force=True)
+        body = data.get("body")
+        if body == None:
+            return jsonify({"msg": "body is required"}), 400
+        post = Post.query.get(post_id)
+        if post == None:
+            return jsonify({"title": "There is no post", "msg": "Post could not updated to db"}),400
+        post.body = body
+        post.save()
+        return jsonify({"title": "Post updated", "msg": "Post updated to db"}), 200
+
+    else: #if request.mehod == 'DELETE'
+        post = Post.query.get(post_id)
+        if post == None:
+            return jsonify({"msg":"No post deleted"}), 400
+        else: 
+            post.delete()
+            return jsonify({"title": "Post deleted", "msg": "Post deleted from db"}), 204
+    
 
+@app.route('/post', methods =['POST'])
+def add_post():
+    data = request.get_json(force=True)
+    body = data.get("body")
+    user_id = data.get("user_id")
+    if body == None:
+        return jsonify({"msg": "body is required"}), 400 #1
+    elif user_id == None:
+        return jsonify({"msg":"user_id required"}), 400  #2
+    else:
+        post = Post(body=body, user_id=user_id)
+        post.save()
+    return jsonify({"title":"Post saved"}),201
+
+
+# Get all posts from db
+@app.route('/posts', methods=['GET'])
+def manage_posts():
+    posts = Post.query.all()
+    return jsonify([{"body":post.body, "user_id":post.user_id, "id": post.id} for post in posts]), 200
 
 
 # Create, Update, Get, Delete one user from db
 @app.route('/user', methods=['POST'])
 def add_user():
     data = request.get_json(force=True)
-    username = data["username"]
-    password = data["password"]
+    username = data.get("username")
+    if username == None:
+        return jsonify({"msg":"username is required"})
+    password = data.get("password")
+    if password == None:
```

Comment:
you can merge `if not username or not password` so you will have only one check.